imap_processing/utils.py

```diff
@@ -0,0 +1,94 @@
+import collections
```

Comment:
It would be good to add a module docstring here, especially since it will likely be used by the whole team

Reply:
Sure thing